src/pages/fieldManagement/settings/defaultFieldComponents/RelationshipEditor.jsx

```diff
@@ -31,7 +31,11 @@ function validateRelationships(relationships) {
   types.forEach(type => {
     const typeLabel = type?.label;
     const roleLabels = get(type, 'roles', []).map(r => r?.label);
-    const emptyRoleLabels = roleLabels.filter(label => label === '');
+    if (roleLabels.length === 0)
+      errors.push(
+        `Each relationship type must have at least one role.`,
+      );
```

Comment:
If two different types each have no roles, this error message is printed twice. Because the error messages are also used as keys, the duplicated messages also cause a key warning.

Reply:
I noticed this. I think the errors (which mention the types explicitly) do a good job clarifying where the end user should be looking for the error(s). As for key warning, maybe make the key the type?

Reply:
The "each relationship type must have at least one role." error prints twice. I favor either only printing that particular message once or adding the type information like you did for the other errors: "Type ${typeLabel} must have at least one role."
<img width="806" alt="duplicate-error-messages" src="https://user-images.githubusercontent.com/50299119/160651810-80c261ae-984e-4e89-8ce4-2c084f7f85c8.png">